components/arm.py

```diff
@@ -172,63 +248,50 @@ def at_goal_extension(self, allowable_error=0.05) -> bool:
         return abs(self.get_extension() - self.goal_extension) < allowable_error
 
     def is_angle_still(self, allowable_speed=0.01) -> bool:
-        """Is the arm currently not moving, allowable speed is in RPS"""
+        """Is the arm currently not moving, allowable speed is in Rotations/s"""
         return abs(self.get_arm_speed()) < allowable_speed
 
-    def brake(self):
-        self.brake_solenoid.set(True)
-
-    def unbrake(self):
+    def brake(self) -> None:
         self.brake_solenoid.set(False)
 
-    def set_target(self, x: float, z: float) -> None:
-        """Set a position in terms of x and y to move the arm to.
-        x: the x position to move the arm to
-        z: the z position to move the arm to
-        """
-        *target, reachable = self.get_arm_from_target(x, z)
-
-        self.set_angle(target[0])
-        self.set_length(target[1])
-
-    def get_arm_from_target(self, x: float, z: float) -> tuple[float, float, bool]:
-        """z (1.3m)
-                        |
-                        |
-        (-1.3m) --------o------- x (1.3m)
-                        |
-                        |
-                        z (-1.3m)
-
-        o is the center of the arm
-
-        x, z: the forward (x) and vertical (z) position to move the arm to in meters reletive
-            to the center of the arm
-
-        returns (
-            radians needed to get to (x,z),
-            arm length needed to get to (x,z)
-            If the position is reachable
-        )
-        If the position is not reachable it clamps it to something that is and gives False
-        """
-        reachable = True
-        # first, get the arm length needed
-        arm_extension = math.sqrt(x**2 + z**2)
-        if arm_extension < self.MIN_EXTENSION or arm_extension > self.MAX_EXTENSION:
-            reachable = False
-            arm_extension = clamp(arm_extension, self.MIN_EXTENSION, self.MAX_EXTENSION)
-
-        # then, get the angle needed from the origin
-        arm_angle = math.atan2(z, x)
-
-        # if the arm can't move to angle_from_origin, return none values
-        if arm_angle > self.MAX_ANGLE or arm_angle < self.MIN_ANGLE:
-            reachable = False
-            arm_angle = clamp(arm_angle, self.MIN_ANGLE, self.MAX_ANGLE)
-
-        return (arm_angle, arm_extension, reachable)
-
-    def on_enable(self):
+    def unbrake(self) -> None:
+        self.brake_solenoid.set(True)
+
+    def on_enable(self) -> None:
         self.extension_controller.reset(self.get_extension())
         self.rotation_controller.reset(self.get_angle())
+
+
+@dataclass
```

Comment:
What dataclass features are being used here? I see our own `__init__` is being defined.